tests/undefined_symbol_in_sdfg_test.py

```diff
@@ -0,0 +1,344 @@
+
+    # Define symbols
+    undefined_dim = symbolic.UndefinedSymbol()
+
+    # Add arrays to SDFG - use the undefined symbol in the shape
+    sdfg.add_array('A', [20, 20], dace.float64)
+    sdfg.add_array('B', [20], dace.float64)
+    sdfg.add_transient('tmp', [20, undefined_dim], dace.float64)
+
+    # Create state
+    state = sdfg.add_state()
+
+    # Create access nodes
+    A = state.add_read('A')
+    tmp = state.add_access('tmp')
+    B = state.add_write('B')
+
+    # Add map
+    map_entry, map_exit = state.add_map('compute', dict(i='0:20', j='0:10'))
+
+    # Add tasklet
+    tasklet = state.add_tasklet('compute', {'a'}, {'b', 't'}, 'b = a; t = a')
+
+    # Connect with edges that use the undefined symbol
+    state.add_edge(A, None, map_entry, None, dace.Memlet())
+    state.add_edge(map_entry, None, tasklet, None, dace.Memlet())
+    state.add_edge(A, None, tasklet, 'a', dace.Memlet('A[i, j]'))
+    state.add_edge(tasklet, 't', tmp, None, dace.Memlet('tmp[i, j]'))
+    state.add_edge(tasklet, 'b', B, None, dace.Memlet('B[i]'))
+    state.add_edge(tasklet, None, map_exit, None, dace.Memlet())
+
+    # Replace undefined symbols in the SDFG with concrete values
+    # This shows how a user would explicitly replace UndefinedSymbol with concrete values
+    replaced_shape = []
+    for dim in sdfg.arrays['tmp'].shape:
+        if isinstance(dim, symbolic.UndefinedSymbol):
+            replaced_shape.append(10)
+        else:
+            replaced_shape.append(dim)
+    sdfg.arrays['tmp'].shape = tuple(replaced_shape)
+
+    # Make sure we can now validate the SDFG
+    sdfg.validate()
+
+
+def test_undefined_symbol_in_argument_validation_failure():
+    """
+    Test that UndefinedSymbol in argument array shapes fails validation,
+    even when only accessed at constant indices.
+    """
+
+    # Create an SDFG manually
+    sdfg = dace.SDFG('undefined_symbol_argument_test')
+
+    # Define symbols
+    undefined_dim = symbolic.UndefinedSymbol()
+
+    # Add arrays to SDFG - use undefined symbol in the input array (not transient)
+    sdfg.add_array('A', [undefined_dim], dace.float64)
+    sdfg.add_array('B', [5], dace.float64)
+
+    # Create state
+    state = sdfg.add_state()
+
+    # Create access nodes
+    A = state.add_read('A')
+    B = state.add_write('B')
+
+    # Add map
+    map_entry, map_exit = state.add_map('compute', dict(i='0:5'))
+
+    # Add tasklet
+    tasklet = state.add_tasklet('compute', {'a'}, {'b'}, 'b = a')
+
+    # Connect edges - only read A at constant index 0
+    state.add_edge(map_entry, None, tasklet, None, dace.Memlet())
+    state.add_edge(A, None, tasklet, 'a', dace.Memlet('A[0]'))
+    state.add_edge(tasklet, 'b', B, None, dace.Memlet('B[i]'))
+    state.add_edge(tasklet, None, map_exit, None, dace.Memlet())
+
+    # This should fail validation because argument array contains undefined symbol
+    with pytest.raises(InvalidSDFGError, match="undefined symbol in dimension"):
+        sdfg.validate()
```

Comment:
Again, this test is a legitimate use case of undefined symbols and should not fail validation. See your previous commits in which this test was called "legitimate". There was a bit of confusion (see above). Please also add sdfg.compile() to make sure it also compiles successfully.